### PR TITLE
 Remove many obsoleted pylint directives

### PR DESCRIPTION
**Pull request by fvkg**
This is made possible by `pylint` recently starting to automatically ignore unused imports inside `if typing.TYPE_CHECKING`, and by the use of variable type annotations (instead of commented type annotations).

### CODE REVIEW

**Comment by vtomole** [comment]
@fvkg 
>  recently starting to automatically ignore unused imports

Which versions of pylint does this?

File: cirq/circuits/text_diagram_drawer.py [hunk 1]
```diff
@@ -60,17 +55,17 @@ class TextDiagramDrawer:
     """
 
     def __init__(self):
-        self.entries = dict()  # type: Dict[Tuple[int, int], _DiagramText]
-        self.vertical_lines = []  # type: List[_VerticalLine]
-        self.horizontal_lines = []  # type: List[_HorizontalLine]
-        self.horizontal_padding = {}  # type: Dict[int, Union[int, float]]
-        self.vertical_padding = {}  # type: Dict[int, Union[int, float]]
+        self.entries: Dict[Tuple[int, int], _DiagramText] = dict()
+        self.vertical_lines: List[_VerticalLine] = []
```
Comment (Strilanc) [inline cirq/circuits/text_diagram_drawer.py]: Mention that you did this in the main PR message.

**Comment by dabacon** [comment]
@viathor the TYPE_CHECKING behavior was added in pylint 2
http://pylint.pycqa.org/en/latest/whatsnew/2.0.html

**Comment by fvkg** [comment]
@dabacon, I think you meant @vtomole.

**Comment by vtomole** [comment]
@dabacon Ah.

**Comment by dabacon** [comment]
Doh yes @vtomole

**Comment by dabacon** [comment]
I can't track down when variable annotations were added but this is part of PEP 526 and python 3.6 so we should be good.

**Pull request merged**